x-pack/plugins/security_solution/server/endpoint/lib/artifacts/task.ts

```diff
@@ -44,22 +48,34 @@ export class ManifestTask {
   constructor(setupContract: ManifestTaskSetupContract) {
     this.endpointAppContext = setupContract.endpointAppContext;
     this.logger = this.endpointAppContext.logFactory.get(this.getTaskId());
+    const { packagerTaskInterval, packagerTaskTimeout } = this.endpointAppContext.serverConfig;
+
+    this.logger.info(
+      `Registering ${ManifestTaskConstants.TYPE} task with timeout of [${packagerTaskTimeout}] and interval of [${packagerTaskInterval}]`
+    );
 
     setupContract.taskManager.registerTaskDefinitions({
       [ManifestTaskConstants.TYPE]: {
         title: 'Security Solution Endpoint Exceptions Handler',
-        timeout: ManifestTaskConstants.TIMEOUT,
+        timeout: packagerTaskTimeout,
```

Comment:
Just a thought if we ever want to do further enhancements in the future - I think we can make this even more resilient if instead of increasing the timeout, we just have the task run a single batch and bump the `runAt` so that it's sooner if there are more batches to be processed. This way, we never risk timeout if there's a truly obscene amount of artifacts and users won't have to tweak `packagerTaskTimeout`. A bit more work though since we'd also need to pass the page through the task state.

Reply:
Thanks for the feedback @joeypoon 
Not sure that would help. The problem is really not the `interval`, but the timeout. Even if we push the `runAt` to be later _(because we detect that task is running longer)_, the `timeout` would still be invoked by Task Manager.  and in this case, I don't even think that Task Manager get the return value from the `run()` (with the `runAt`) because it has stop monitoring it.

One of the things I'm looking into for `main` is to possibly implement the `cancel()` callback of the task registration so that we actually cancel the processing (today, when Task Manager detects a `timeout` condition, it "cancels" the running task by calling the `cancel()` callback, but as you can see below (line 97) we do nothing with it (I added a logging message below just so that we have something in the logs).

Reply:
Sorry, `bump` was a poor choice of word. What I mean is not to make `runAt` later but rather immediately if more batches remain. That way we keep each individual task run at a known quantity of artifacts. So:
1) run batch
2) if more batches, run again immediately
3) if no more batches, schedule for regular interval

Reply:
I guess I'm not following. What do you mean by Batches?